crates/bevy_winit/src/state.rs

```diff
@@ -149,6 +149,8 @@ pub struct CustomCursorCache(pub HashMap<CustomCursorCacheKey, winit::window::Cu
 /// A source for a cursor. Is created in `bevy_render` and consumed by the winit event loop.
 #[derive(Debug)]
 pub enum CursorSource {
+    /// A hidden cursor, used to hide the cursor in `winit_window`
+    Hidden,
```

Comment:
I don't think we should propagate cursor visibility to `bevy_winit`. Custom cursors are required to be changed in such a convoluted way because they require access to the winit event loop. Cursor visibility can be changed directly with `NonSendMut<WinitWindows>` in any system just like it was done before with `cursor_options`.

Reply:
I'm starting to think what we really need to do is do something along the lines of not having a cursor from being instead `Hidden` to just not having a cursor source or `None`. This clearly denotes the feature set each cursor source has - the `SystemCursor` is heavily part of the `winit` package we use, hence the related feature set. Now that we have `CursorSource::Custom`, if we think that over time it ends up having a different feature set, maybe something like if we were to support gifs as cursors, or sprite sheets as cursors with animations, there might be options relating how fast to play the gif or whether or not to loop it's animation. This way, most of the examples will look almost the same in functionality, just changing a bool

```
pub enum CursorSource {
    None,
    Custom{ custom_cursor: CustomCursor, visibility: bool },
    System{ system_cursor: SystemCursor, visibility: bool },
}
```

Reply:
As to the 

> Cursor visibility can be changed directly with `NonSendMut<WinitWindows>` in any system just like it was done before with `cursor_options`.

I don't get quite this means? Are you suggesting that if it isn't a `CustomCursor` then we should just use the system `winit` related API directly? I think we were initially against that because we would have two API's and it could get confusing, so I would like your take on it.

Reply:
What I would be proposing is to just move cursor options from the window component to the cursor component and doing absolutely everything the exact same it was done before and make a new cursor component.
```rust
struct Cursor {
    pub visible: bool,
    ... other options ...
    pub icon: CursorIcon,
}
```
Then move everything that was done in the window update system to the cursor update system. 

But I guess that is a different proposal than what is included in the issue, so we can also implement this proposal if others like it better.


> I don't get quite this means? Are you suggesting that if it isn't a `CustomCursor` then we should just use the system `winit` related API directly? I think we were initially against that because we would have two API's and it could get confusing, so I would like your take on it.

Yes. `CursorSource` is an implementation detail for smuggling cursor icon bytes to the `bevy_winit` crate. It would lead to less spread out code and less complexity to just set the `WinitWindows` cursor visibility boolean in the `bevy_render` cursor system. It doesn't create any additional APIs. Functionally your code does the same thing so consider what would be simpler.

Reply:
> Now that we have CursorSource::Custom, if we think that over time it ends up having a different feature set.

`CursorSource` is a direct equivalent to `winit`s native cursor sources. Those are clearly defined and not flexible, so the `CursorSource` doesn't need to be flexible.

E.g. if we want to use sprite sheets as animated cursors, those would need to be converted to be `winit` compatible outside of `bevy_winit` anyway because in can't use assets.